hook!-full-game/src/Combat/HitBox/HitBox.gd

```diff
@@ -10,7 +10,7 @@ func _ready() -> void:
 
 
 func _on_area_entered(damage_source: Area2D) -> void:
-	var hit = HitFactory.calculate_hit(damage_source)
+	var hit = HitFactory.generate_hit(damage_source)
```

Comment:
I don't think `HitFactory` adds anything useful. Since `Hit` is already declared with `class_name`, we can already do `Hit.new([...])` wherever needed and the naming is pretty explanatory.

Reply:
Right now it doesn't do much, yes. Having this is mainly for OOP and code structure, having a middle man between an attacker and a target that's responsible for transforming input stats from e.g. an enemy and its equipment into an encapsulated hit that represents damage and inflicted status effects.

But yes, in this game it's probably unnecessary.